src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1225,8 +1225,12 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
 
             // Dev11 searches for the key file in the current directory and assembly output directory.
             // We always look to base directory and then examine the search paths.
-            keyFileSearchPaths.Add(baseDirectory);
-            if (baseDirectory != outputDirectory)
+            if (!string.IsNullOrWhiteSpace(baseDirectory))
+            {
+                keyFileSearchPaths.Add(baseDirectory);
+            }
+
+            if (!string.IsNullOrWhiteSpace(outputDirectory) && baseDirectory != outputDirectory)
```

Comment:
Need a test for when `baseDirectory` is `null` but a valid `outputDirectory` is specified. Should ensure it ends up in the options.